tasks/post_to_catalog/post_to_catalog.py

```diff
@@ -0,0 +1,241 @@
+                        "name": file["name"],
+                        "cumulus_archive_location": file["cumulusArchiveLocation"],
+                        "orca_archive_location": file["orcaArchiveLocation"],
+                        "key_path": file["keyPath"],
+                        "size_in_bytes": file["sizeInBytes"],
+                        "hash": file.get("hash", None),
+                        "hash_type": file.get("hashType", None),
+                        "version": file["version"],
+                        "ingest_time": file["ingestTime"],
+                        "etag": file["etag"],
+                    }
+                )
+            if any(file_parameters):
+                LOGGER.debug("Creating file records.")
+                connection.execute(create_file_sql(), file_parameters)
+    except Exception as sql_ex:
+        # Can't use f"" because of '{}' bug in CumulusLogger.
+        LOGGER.error(
+            "Error while posting provider '{provider_id}', collection '{collection_id}', "
+            "granule '{cumulus_granule_id}' to inventory: {sql_ex}",
+            provider_id=provider["providerId"],
+            collection_id=collection["collectionId"],
+            cumulus_granule_id=granule["cumulusGranuleId"],
+            sql_ex=sql_ex,
+        )
+        raise
+
+
+def create_provider_sql():
+    return text(
+        """
+        INSERT INTO providers
+            ("provider_id", "name")
+        VALUES
+            (:provider_id, :name)
+        ON CONFLICT DO NOTHING
+        """
+    )
+
+
+def create_collection_sql():
+    return text(
+        """
+    INSERT INTO collections
+            ("collection_id", "shortname", "version")
+    VALUES
+        (:collection_id, :shortname, :version)
+    ON CONFLICT DO NOTHING
+    """
+    )
+
+
+def create_granule_sql():
+    return text(
+        """
+    INSERT INTO granules
+        ("collection_id", "cumulus_granule_id", "execution_id", "ingest_time", "cumulus_create_time", "last_update")
+    VALUES
+        (:collection_id, :cumulus_granule_id, :execution_id, :ingest_time, :cumulus_create_time, :last_update)
+    ON CONFLICT ("collection_id", "cumulus_granule_id") DO UPDATE
+        SET "execution_id"=:execution_id, "last_update"=:last_update
+    RETURNING id"""
+    )
+    # ON CONFLICT will only trigger if both collection_id and cumulus_granule_id match.
+
+
+def create_file_sql():
+    return text(
+        """
+    INSERT INTO files
+        ("granule_id", "name", "orca_archive_location", "cumulus_archive_location", "key_path", "ingest_time", "etag", 
+        "version", "size_in_bytes", "hash", "hash_type")
+    VALUES
+        (:granule_id, :name, :orca_archive_location, :cumulus_archive_location, :key_path, :ingest_time, :etag, 
+        :version, :size_in_bytes, :hash, :hash_type)
+    ON CONFLICT ("cumulus_archive_location", "key_path") DO UPDATE
+        SET "granule_id"=:granule_id, "name"=:name, "orca_archive_location"=:orca_archive_location, 
```

Comment:
I don't think the logic for this upsert is correct looking at the file table definition. Looking at the table definition, we have 2 unique constraints seen below.
```
CONSTRAINT UNIQUE_orca_archive_location_key_path UNIQUE (orca_archive_location, key_path)
CONSTRAINT UNIQUE_cumulus_archive_location_key_path UNIQUE (cumulus_archive_location, key_path)
```
I feel your choice for the inferred `UNIQUE_cumulus_archive_location_key_path` is correct, but we probably don't want to to change the FK value. Possibly change to something more like the following?

```
ON CONFLICT (cumulus_archive_location, key_path) DO UPDATE
        SET 
            name= EXCLUDED.name, -- not sure if this would really change given the key
            orca_archive_location=EXCLUDED.orca_archive_location,
            ingest_time=EXCLUDED.ingest_time, 
            etag=EXCLUDED.etag, 
            version=EXCLUDED.version, 
            size_in_bytes=EXCLUDED.size_in_bytes, 
            hash=EXCLUDED.hash, 
            hash_type=EXCLUDED.hash_type         
"""
```

I also see that a lot of your columns are quoted and they shouldn't need to be.

Reply:
- So we will not support moving a file from one granule to another? Definitely not something I'd anticipate often, but I want the exclusion to be explicitly acknowledged.
- Any particular reason you're using `EXCLUDED` instead of the parameter names? I feel that's not a very clear keyword.
- Removed column quotes.